s3/s3_transport/unit_tests/src/test_s3_transport.cpp

```diff
@@ -0,0 +1,995 @@
+                          int thread_count,
+                          std::ios_base::openmode open_modes = std::ios_base::in | std::ios_base::out)
+{
+    std::string access_key, secret_access_key;
+    read_keys(keyfile, access_key, secret_access_key);
+
+    read_write_stage_and_cleanup(bucket_name, filename, object_prefix);
+
+    std::string comparison_filename = filename + ".comparison";
+
+    irods::thread_pool writer_threads{thread_count};
+
+    for (int thread_number = 0; thread_number <  thread_count; ++thread_number) {
+
+        irods::thread_pool::post(writer_threads, [bucket_name, access_key,
+                secret_access_key, filename, object_prefix, thread_count, thread_number,
+                comparison_filename, open_modes] () {
+
+
+            read_write_on_file(hostname.c_str(), bucket_name.c_str(), access_key.c_str(), secret_access_key.c_str(),
+                    filename.c_str(), object_prefix.c_str(), thread_count, thread_number, comparison_filename.c_str(),
+                    open_modes);
+        });
+    }
+
+    writer_threads.join();
+
+    check_read_write_results(bucket_name, filename, object_prefix);
+}
+
+
+TEST_CASE("quick test upload", "[quick_test][quick_test_upload]")
+{
+    rodsLogLevel(log_level);
+
+    SECTION("upload large file with multiple threads")
+    {
+        int thread_count = 7;
+        std::string filename = "large_file";
+        std::string object_prefix = "dir1/dir2/";
+        bool expected_cache_flag = false;
+        do_upload_thread(bucket_name, filename, object_prefix, keyfile, thread_count, expected_cache_flag);
+    }
+}
+
+TEST_CASE("quick test download", "[quick_test][quick_test_download]")
+{
+    rodsLogLevel(log_level);
+
+    SECTION("download large file with multiple threads")
+    {
+        int thread_count = 7;
+        std::string filename = "large_file";
+        std::string object_prefix = "dir1/dir2/";
+        bool expected_cache_flag = false;
+        do_download_thread(bucket_name, filename, object_prefix, keyfile, thread_count, expected_cache_flag);
+    }
+}
+
+/*TEST_CASE("shmem tests", "[shmem]")
+{
+
+    SECTION("test shmem with named mutex already locked")
+    {
+        using constants = irods::experimental::io::s3_transport::constants;
+        namespace bi = boost::interprocess;
+
+        // lock the named mutex and leave locked, will time out and shmem will be
+        // created succcessfully
+        std::string object_key = "dir1/dir2/large_file";
+        std::string shmem_key = constants::SHARED_MEMORY_KEY_PREFIX +
+                std::to_string(std::hash<std::string>{}(object_key));
+
+        bi::named_mutex mutex(bi::open_or_create, shmem_key.c_str());
+        mutex.lock();
+
+        int thread_count = 7;
+        std::string filename = "large_file";
+        std::string object_prefix = "dir1/dir2/";
+        do_upload_thread(bucket_name, filename, object_prefix, keyfile, thread_count);
+    }
+}*/
```

Comment:
remove?    i see shmem tests 2 just below...

Reply:
Done

Reply:
This is removed